flang/lib/Lower/OpenMP/DataSharingProcessor.cpp

```diff
@@ -289,12 +290,34 @@ void DataSharingProcessor::collectSymbolsInNestedRegions(
        eval.getNestedEvaluations()) {
     if (nestedEval.hasNestedEvaluations()) {
       if (nestedEval.isConstruct())
-        // Recursively look for OpenMP constructs within `nestedEval`'s region
```

Comment:
I would leave this comment. It helps to understand what this part of the code is doing, that we only continue looking for OpenMP constructs recursively for non-OpenMP constructs. Otherwise `converter.collectSymbolSet` can be used to collect the remaining symbols.

Reply:
Not sure why I removed. Thanks for noticing.